test/module/irohad/multi_sig_transactions/state_test.cpp

```diff
@@ -25,228 +13,338 @@ using namespace std;
+      makeTestBatch(txBuilder(1, time, quorum)), 0, makeSignature("2", "2"));
+  ASSERT_EQ(0, state_after_one_tx.getBatches().size());
 
-  auto state_after_three_txes = state += makeTx(1, time, makeKey(), quorum);
-  ASSERT_EQ(1, state_after_three_txes.getTransactions().size());
-  ASSERT_EQ(0, state.getTransactions().size());
+  auto state_after_three_txes = state += addSignatures(
+      makeTestBatch(txBuilder(1, time, quorum)), 0, makeSignature("3", "3"));
+  ASSERT_EQ(1, state_after_three_txes.getBatches().size());
+  ASSERT_EQ(0, state.getBatches().size());
 }
 
+/**
+ * @given two states with same transaction, where transaction will complete
+ * after merge
+ * @when  merge states
+ * @then  checks that transaction go out to completed state
+ */
 TEST(StateTest, UpdateStateWithNewStateUntilQuorum) {
-  log_->info("Merge two states that contains common transaction");
-
   auto quorum = 3u;
   auto keypair = makeKey();
   auto time = iroha::time::now();
 
   auto state1 = MstState::empty();
-  state1 += makeTx(1, time, makeKey(), quorum);
-  state1 += makeTx(1, time, keypair, quorum);
-  state1 += makeTx(2, time, makeKey(), quorum);
-  ASSERT_EQ(2, state1.getTransactions().size());
+  state1 += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                          0,
+                          makeSignature("1_1", "1_1"));
+  state1 += addSignatures(
+      makeTestBatch(txBuilder(2, time)), 0, makeSignature("2", "2"));
+  state1 += addSignatures(
+      makeTestBatch(txBuilder(2, time)), 0, makeSignature("3", "3"));
+  ASSERT_EQ(2, state1.getBatches().size());
 
   auto state2 = MstState::empty();
-  state2 += makeTx(1, time, keypair, quorum);
-  state2 += makeTx(1, time, makeKey(), quorum);
-  ASSERT_EQ(1, state2.getTransactions().size());
+  state2 += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                          0,
+                          makeSignature("1_2", "1_2"));
+  state2 += addSignatures(makeTestBatch(txBuilder(1, time, quorum)),
+                          0,
+                          makeSignature("1_3", "1_3"));
+  ASSERT_EQ(1, state2.getBatches().size());
 
   auto completed_state = state1 += state2;
-  ASSERT_EQ(1, completed_state.getTransactions().size());
-  ASSERT_EQ(1, state1.getTransactions().size());
+  ASSERT_EQ(1, completed_state.getBatches().size());
+  ASSERT_EQ(1, state1.getBatches().size());
 }
 
+/**
+ * Tests expired completer, which checks that all transactions in batch are not
+ * expired
+ */
 class TimeTestCompleter : public iroha::DefaultCompleter {
-  bool operator()(const DataType &tx, const TimeType &time) const override {
-    return tx->createdTime() < time;
+  bool operator()(const DataType &batch, const TimeType &time) const override {
+    return std::all_of(
+        batch->transactions().begin(),
+        batch->transactions().end(),
+        [&time](const auto &tx) { return tx->createdTime() < time; });
   }
 };
 
+/**
+ * @given state with one transaction
```

Comment:
-> given a timepoint AND a state with an expired transaction
when erase by time is called
then the resulting state contains the expired transaction